code/obj/item/device/radios/intercoms.dm

```diff
@@ -7,14 +7,31 @@
 #endif
 	anchored = 1.0
 	plane = PLANE_NOSHADOW_ABOVE
-	mats = 0
+	mats = 3
+	deconstruct_flags = DECON_SCREWDRIVER | DECON_WRENCH | DECON_WIRECUTTERS | DECON_MULTITOOL
 	chat_class = RADIOCL_INTERCOM
 	var/number = 0
 	rand_pos = 0
 	desc = "A wall-mounted radio intercom, used to communicate with the specified frequency. Usually turned off except during emergencies."
 	hardened = 0
 
+/obj/item/device/radio/intercom/proc/change_dir(obj/item/AM, old_dir, new_dir)
```

Comment:
I would change this proc name to `update_position_from_dir` or something similar. `update_dir` is very generic and not meaningfully different from `set_dir`, so I think this would help. For example, calling "change_dir" in `New` doesn't seem quite right, but calling `update_position_from_dir` explains what it's doing better.